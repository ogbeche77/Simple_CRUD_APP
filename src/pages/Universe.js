import React, { useState, useEffect } from "react";
import { useHistory, useParams } from "react-router-dom";
import { Card, Image, Button, Label } from "semantic-ui-react";
import {
  cyInputUniverseName,
  cyInputMaximumSize,
  cyInputCurrentSize,
  cyAddNewUniverseButton,
} from "../handles/index.js";

const Universe = () => {
  const { id } = useParams;
  const [universe, setUniverse] = useState(null);
  const [loading, setLoading] = useState(true);
  const [newUniverse, setNewUniverse] = useState("");
  const [maximumSize, setMaximumSize] = useState("");
  const [currentSize, setCurrentSize] = useState("");
  const history = useHistory();

  useEffect(() => {
    fetch("http://localhost:8080/universes")
      .then((res) => {
        return res.json();
      })
      .then((data) => {
        setUniverse(data);
        setLoading(false);
      })
      .catch((err) => {
        console.log(err);
      });
  }, [id]);
  const handleSumit = (e) => {
    e.preventDefault();
    const createNewUniverse = { newUniverse, maximumSize, currentSize };

    fetch("http://localhost:8080/universes", {
      method: "POST",
      headers: { "Content-Type": "application/json" },
      body: JSON.stringify(createNewUniverse),
    }).then(() => {
      history.go(0);
    });
  };

  const handleDelete = () => {
    fetch("http://localhost:8080/universes/" + id, {
      method: "DELETE",
    }).then(() => {
      history.go(0);
    });
  };

  return (
    <div data-testid="universe">
      <Card fluid>
        <Card.Content>
          <Image
            floated="right"
            size="mini"
            src="https://react.semantic-ui.com/images/avatar/large/molly.png"
          />

          <Card.Description>
            {loading && <h1>Loading</h1>}
            {universe &&
              universe.map((universes, index) => {
                return (
                  <div key={index}>
                    <h1> {universes.newUniverse} </h1>
                    <div className="universe-body">
                      <Label basic color="grey" pointing="left">
                        Maximum Size: {universes.maximumSize}
                      </Label>

                      <Label basic color="grey" pointing="left">
                        Current Size: {universes.currentSize}
                      </Label>
                      <Button
                        onClick={handleDelete}
                        type="submit"
                        color="brown"
                      >
                        {" "}
                        Delete
                      </Button>
                    </div>
                  </div>
                );
              })}
          </Card.Description>
        </Card.Content>
        <Card.Content extra>
          <form className="form-container" onSubmit={handleSumit}>
            <h3 className="form-title">Add a universe</h3>
            <div className="form-content">
              <input
                placeholder="Add a universe"
                type="text"
                required={true}
                value={newUniverse}
                onChange={(e) => setNewUniverse(e.target.value)}
                data-cy={cyInputUniverseName}
              />
              <input
                placeholder="Add Maximum Size"
                type="number"
                required={true}
                value={maximumSize}
                onChange={(e) => setMaximumSize(e.target.value)}
                data-cy={cyInputMaximumSize}
              />
              <input
                placeholder="Add Current Size"
                type="number"
                required={true}
                value={currentSize}
                onChange={(e) => setCurrentSize(e.target.value)}
                data-cy={cyInputCurrentSize}
              />
            </div>
            <Button
              type="submit"
              color="brown"
              data-cy={cyAddNewUniverseButton}
            >
              {" "}
              Submit
            </Button>
          </form>
        </Card.Content>
      </Card>
    </div>
  );
};

export default Universe;
