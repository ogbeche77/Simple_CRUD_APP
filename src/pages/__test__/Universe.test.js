import React from "react";
import ReactDOM from "react-dom";
import Universe from "../Universe";
import { render, screen, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom";
import renderer from "react-test-renderer";

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Universe></Universe>, div);
});

it("renders universe correctly", () => {
  const { getByTestId } = render(<Universe></Universe>);
  expect(getByTestId("universe")).toHaveTextContent("Add a universe");
});

it("matches universe component snapshot", () => {
  const universeSnapshot = renderer.create(<Universe></Universe>).toJSON();
  expect(universeSnapshot).toMatchSnapshot();
});
