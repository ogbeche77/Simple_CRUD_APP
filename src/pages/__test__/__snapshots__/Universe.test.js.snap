// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches universe component snapshot 1`] = `
<div
  data-testid="universe"
>
  <div
    className="ui fluid card"
    onClick={[Function]}
  >
    <div
      className="content"
    >
      <img
        className="ui mini right floated image"
        src="https://react.semantic-ui.com/images/avatar/large/molly.png"
      />
      <div
        className="description"
      >
        <h1>
          Loading
        </h1>
      </div>
    </div>
    <div
      className="extra content"
    >
      <form
        className="form-container"
        onSubmit={[Function]}
      >
        <h3
          className="form-title"
        >
          Add a universe
        </h3>
        <div
          className="form-content"
        >
          <input
            data-cy="InputUniverseName"
            onChange={[Function]}
            placeholder="Add a universe"
            required={true}
            type="text"
            value=""
          />
          <input
            data-cy="InputMaximumSize"
            onChange={[Function]}
            placeholder="Add Maximum Size"
            required={true}
            type="number"
            value=""
          />
          <input
            data-cy="InputCurrentSize"
            onChange={[Function]}
            placeholder="Add Current Size"
            required={true}
            type="number"
            value=""
          />
        </div>
        <button
          className="ui brown button"
          data-cy="AddNewUniverseButton"
          onClick={[Function]}
          type="submit"
        >
           
          Submit
        </button>
      </form>
    </div>
  </div>
</div>
`;
